addons/stock/wizard/product_replenish_views.xml

```diff
@@ -11,33 +11,45 @@
                 a manufacturing order or a transfer.
                 </p>
                 <group>
-                    <field name="product_tmpl_id" invisible="1"/>
-                    <field name="product_has_variants" invisible="1"/>
-                    <field name="product_id"
-                        domain="[('product_tmpl_id', '=', product_tmpl_id)]"
-                        attrs="{'readonly': [('product_has_variants', '=', False)]}"
-                        options="{'no_create_edit':1}"/>
-                    <field name="product_uom_category_id" invisible="1"/>
-                    <label for="quantity"/>
-                    <div class="o_row">
-                        <field name="quantity" />
-                        <field name="product_uom_id"
-                            domain="[('category_id', '=', product_uom_category_id)]"
-                            groups="uom.group_uom"/>
-                    </div>
-                    <field name="date_planned"/>
-                    <field name="warehouse_id"
-                        groups="stock.group_stock_multi_warehouses"/>
-                    <field name="route_ids"
-                        widget="many2many_tags"/>
-                    <field name="company_id" invisible="1"/>
+                    <group>
+                        <field name="product_tmpl_id" invisible="1"/>
+                        <field name="product_has_variants" invisible="1"/>
+                        <field name="product_id"
+                            domain="[('product_tmpl_id', '=', product_tmpl_id)]"
+                            attrs="{'readonly': [('product_has_variants', '=', False)]}"
+                            options="{'no_create_edit':1}"/>
+                        <field name="product_uom_category_id" invisible="1"/>
+                        <label for="quantity"/>
+                        <div class="o_row">
+                            <field name="quantity" />
+                            <field name="product_uom_id"
+                                domain="[('category_id', '=', product_uom_category_id)]"
+                                groups="uom.group_uom"/>
+                        </div>
+                        <label for="forecasted_quantity"/>
+                        <div class="o_row">
+                            <field name="forecasted_quantity" readonly="1"/>
+                            <field name="product_uom_id" domain="[('category_id', '=', product_uom_category_id)]" groups="uom.group_uom" readonly="1"/>
```

Comment:
You shouldn't reuse the same field here, because if you change the quantity's UoM in the wizard, it will change the forecast's UoM too, which doesn't make sense (also, you will have issue with the `readonly` thing).
I suggest you to use two fields:
- `product_uom_id` which will be a related field from `product_id.uom_id` and displayed beside the forecast quantity;
- `uom_id` which will be an editable field (keep the domain on the category) used for the replenish wizard's quantity.